diff --git lineedit/keymap.c lineedit/keymap.c
index 88a8110..3ea2b18 100644
--- lineedit/keymap.c
+++ lineedit/keymap.c
@@ -78,6 +78,7 @@ void le_keymap_init(void)
     Set(Key_end,       cmd_end_of_line);
     Set(Key_c_j,       cmd_accept_line);
     Set(Key_c_m,       cmd_accept_line);
+    Set(Key_enter,     cmd_accept_line);
     Set(Key_interrupt, cmd_abort_line);
     Set(Key_c_c,       cmd_abort_line);
     Set(Key_eof,       cmd_eof_if_empty);
@@ -114,6 +115,7 @@ void le_keymap_init(void)
     Set(L"0",          cmd_bol_or_digit);
     Set(Key_c_j,       cmd_accept_line);
     Set(Key_c_m,       cmd_accept_line);
+    Set(Key_enter,     cmd_accept_line);
     Set(Key_interrupt, cmd_abort_line);
     Set(Key_c_c,       cmd_abort_line);
     Set(Key_eof,       cmd_eof_if_empty);
@@ -228,6 +230,7 @@ void le_keymap_init(void)
     Set(Key_escape "-",     cmd_digit_argument);
     Set(Key_c_j,            cmd_accept_line);
     Set(Key_c_m,            cmd_accept_line);
+    Set(Key_enter,          cmd_accept_line);
     Set(Key_interrupt,      cmd_abort_line);
     Set(Key_c_c,            cmd_abort_line);
     Set(Key_eof,            cmd_eof_or_delete);
@@ -302,6 +305,7 @@ void le_keymap_init(void)
     t = trie_create();
     Set(Key_c_v,       cmd_expect_verbatim);
     Set(Key_c_m,       cmd_accept_line);
+    Set(Key_enter,     cmd_accept_line);
     Set(Key_interrupt, cmd_abort_line);
     Set(Key_c_c,       cmd_abort_line);
     Set(Key_c_l,       cmd_redraw_all);
